mrr/src/main/java/gov/cdc/dex/mrr/TimerTriggerFunction.kt

```diff
@@ -20,49 +21,51 @@ import java.time.LocalDateTime
     ) {
         context.logger.info("PhinVocabRead time trigger processed a request.")
-        var jedis: Jedis? = null
+
            try {
-                jedis = RedisUtility().redisConnection()
-
-            // Simple PING command
-            context.logger.info("\nCache Command  : Ping")
-               if (jedis != null) {
-                   context.logger.info("Cache Response : " + jedis.ping())
-                   val client = VocabClient()
-                   context.logger.info("STARTING VocabClient services")
-                   val valueSets = client.getAllValueSets() as List<ValueSet>?
-
-                   valueSets?.let { context.logger.info("Count of ValueSets:  ${valueSets.size}") }
-
-                   val vocabMap: MutableMap<StringBuilder, Any> = HashMap()
-                   val gson = GsonBuilder().create()
-                   for (e in valueSets!!) {
-                       val valueSetConcepts = client.getValueSetConcepts(e) as List<ValueSetConcept>?
-                       val key = client.getValueSetKey(e)
-                       vocabMap[key] = gson.toJson(valueSetConcepts)
-                       if (jedis.exists(key.toString()))
-                           jedis.del(key.toString())
-                       jedis.set(key.toString(), gson.toJson(valueSetConcepts))
+               val client = VocabClient()
+               val exe  = Executors.newCachedThreadPool()
+               context.logger.info("STARTING VocabClient services")
+               val valueSets = client.getAllValueSets() as List<ValueSet>?
+
+               if (valueSets != null) {
+                   context.logger.info("Count of ValueSets:  ${valueSets.size}")
+               }
+               var vsCount =0
+               var valueSetConcepts = listOf<ValueSetConcept>()
+
+               for (e in valueSets!!) {
+                   vsCount += 1
+                   val key = client.getValueSetKey(e)
+                   if(client.getValueSetConcepts(e) != null) {
+                       valueSetConcepts = (client.getValueSetConcepts(e) as List<ValueSetConcept>?)!!
+                   }
+
+                   exe.submit {
+                       //  context.logger.info("KEY: ${key.toString()} ")
+                       // context.logger.info("KEY count: ${vsCount} ")
+                       client.setValueSetConcepts(valueSetConcepts,key.toString())
                    }
-                   context.logger.info("END OF VocabClient services")
+
+                   context.logger.info("Key: ${key.toString()} + count of ${vsCount}")
+
 
                }
-               else
-                context.logger.info("Failure in getting  Radis connection ")
+               context.logger.info("END OF VocabClient services")
         } catch(e:Exception){
-            context.logger.log(error("Failure in PhinVocabRead function : ${e.printStackTrace()} "))
+            context.logger.info("Failure in PhinVocabRead function : ${e.printStackTrace()} ")
```

Comment:
why downgrading this to info? shouldn't this be an error?

Also, not sure we need to use context.logger - we can still use our own loggers (but i think is fine to use context.logger) - just don't want to be passing context as parameters to other classes.)

Reply:
needs some testing on portal, didn't updated it.